ds3/ds3.py

```diff
@@ -500,11 +514,7 @@ def putBulk(self, bucketName, fileInfoList):
         `objectName` does not have to be the actual name on the local file system, but it will be the name that you must
         initiate a single object put to later.  `size` must reflect the actual size of the file that is being put.
         '''
-        bulkObjs = libds3.lib.ds3_init_bulk_object_list(len(fileInfoList))
-        bulkObjsList = bulkObjs.contents.list
-        for i in xrange(0, len(fileInfoList)):
-            bulkObjsList[i].name = libds3.lib.ds3_str_init(fileInfoList[i][0])
-            bulkObjsList[i].length = fileInfoList[i][1]
+        bulkObjs = libds3.toDs3BulkObjectList(typeCheckObjectList(fileInfoList))

```

Comment:
I went to change this, but realized toDs3BulkObjectList needs an object with a length passed so it can allocate enough memory, which means we can't use a generator here.